crates/templates/src/manager.rs

```diff
@@ -1387,6 +1403,104 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn cannot_generate_over_existing_files_by_default() {
+        let temp_dir = tempdir().unwrap();
+        let store = TemplateStore::new(temp_dir.path());
+        let manager = TemplateManager { store };
+        let source = TemplateSource::File(project_root());
+
+        manager
+            .install(&source, &InstallOptions::default(), &DiscardingReporter)
+            .await
+            .unwrap();
+
+        let template = manager.get("http-rust").unwrap().unwrap();
+
+        let dest_temp_dir = tempdir().unwrap();
+        let output_dir = dest_temp_dir.path().join("myproj");
+
+        tokio::fs::create_dir_all(&output_dir).await.unwrap();
+        let manifest_path = output_dir.join("spin.toml");
+        tokio::fs::write(&manifest_path, "cookies").await.unwrap();
+
+        let values = [
+            ("project-description".to_owned(), "my desc".to_owned()),
+            ("http-path".to_owned(), "/path/...".to_owned()),
+        ]
+        .into_iter()
+        .collect();
+        let options = RunOptions {
+            variant: crate::template::TemplateVariantInfo::NewApplication,
+            output_path: output_dir.clone(),
+            name: "my project".to_owned(),
+            values,
+            accept_defaults: false,
+            no_vcs: false,
+            allow_overwrite: false,
+        };
+
+        template
+            .run(options)
+            .silent()
+            .await
+            .expect_err("generate into existing dir should have failed");
+
+        assert!(tokio::fs::read_to_string(&manifest_path)
+            .await
+            .unwrap()
+            .contains("cookies"));
+    }
+
+    #[tokio::test]
+    async fn can_generate_over_existing_files_if_so_configured() {
```

Comment:
nit: there's a lot of set up for these tests that is indentical. Would it make sense to test both scenarios in the same test perhaps by doing the "cookies" test, then deleting the spin.toml, and trying again?

Reply:
Yes, there's a lot of shared test setup across the template tests.  I'd prefer to keep the individual tests specific, but as a later pass we should refactor the entire test suite to reduce duplication/copy-paste and make the "what is specific about this setup and about the expected results" pop a lot more.  So I am going to punt on this for now but I do totally accept your point!